lib/core/jobs/consumer.rb

```diff
@@ -0,0 +1,120 @@
+
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+module Core
+  module Jobs
+    class Consumer
+      def initialize(poll_interval: 3, termination_timeout: 60, min_threads: 1, max_threads: 5, max_queue: 100, idle_time: 5)
+        @poll_interval = poll_interval
+        @termination_timeout = termination_timeout
+        @min_threads = min_threads
+        @max_threads = max_threads
+        @max_queue = max_queue
+        @idle_time = idle_time
+
+        @running = Concurrent::AtomicBoolean.new(false)
+
+        Kernel.at_exit { shutdown! }
+      end
+
+      def subscribe!(index_name:)
+        @index_name = index_name
+
+        start_loop!
+      end
+
+      def running?
+        # @TODO check if a loop thread is alive
+        pool.running? && @running.true?
+      end
+
+      def shutdown!
+        Utility::Logger.info("Shutting down consumer for #{@index_name} index")
+        @running.make_false
+        pool.shutdown
+        pool.wait_for_termination(@termination_timeout)
+        # reset pool
+        @pool = nil
+      end
+
+      private
+
+      def start_loop!
+        Utility::Logger.info("Starting a new consumer for #{@index_name} index")
+
+        Thread.new do
+          # assign a name to the thread
+          # see @TODO in #self.running?
+          Thread.current[:name] = "consumer-group-#{@index_name}"
+
+          loop do
+            if @running.false?
+              Utility::Logger.info('Shutting down the loop')
+              break
+            end
+
+            sleep(@poll_interval)
+            Utility::Logger.debug('Getting registered connectors')
+
+            connectors = ready_for_sync_connectors
+            next unless connectors.any?
+
+            Utility::Logger.debug("Number of available connectors: #{connectors.size}")
+
+            # @TODO It is assumed that @index_name is used to retrive pending jobs.
+            # This will be discussed after 8.6 release
+            pending_jobs = Core::ConnectorJob.pending_jobs(connectors_ids: connectors.keys)
+            Utility::Logger.info("Number of pending jobs: #{pending_jobs.size}")
+
+            pending_jobs.each do |job|
+              connector_settings = connectors[job.connector_id]
+
+              pool.post do
+                Core::ElasticConnectorActions.ensure_content_index_exists(connector_settings.index_name)
+                job_runner = Core::SyncJobRunner.new(connector_settings, job)
+                job_runner.execute
```

Comment:
Just to double check, is there a log line somewhere that says that connector [MONGO] picked up job [12safkjslat1]? If no, nice to add here